src/aws-fsx/ontap-file-system.ts

```diff
@@ -0,0 +1,514 @@
+    }
+  }
+
+  private validateEndpointIpAddressRange(deploymentType: OntapDeploymentType, endpointIpAddressRange?: string): void {
+    if (endpointIpAddressRange == null || Token.isUnresolved(endpointIpAddressRange)) {
+      return;
+    }
+    if (deploymentType !== OntapDeploymentType.MULTI_AZ_1 && deploymentType !== OntapDeploymentType.MULTI_AZ_2) {
+      throw new Error("'endpointIpAddressRange' can only be specified for multi-AZ file systems");
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{9,17}$/.test(endpointIpAddressRange)) {
+      throw new Error(
+        "'endpointIpAddressRange' must be between 9 and 17 characters long and not contain any of the following characters: \\u0000, \\u0085, \\u2028, \\u2029, \\r, or \\n",
+      );
+    }
+  }
+
+  private validateFsxAdminPassword(fsxAdminPassword?: string): void {
+    if (fsxAdminPassword == null || Token.isUnresolved(fsxAdminPassword)) {
+      return;
+    }
+    if (!/^[^\u0000\u0085\u2028\u2029\r\n]{8,50}$/.test(fsxAdminPassword)) {
+      throw new Error(
+        "'fsxAdminPassword' must be between 8 and 50 characters long and not contain any of the following characters: \\u0000, \\u0085, \\u2028, \\u2029, \\r, or \\n",
+      );
+    }
+    // must contain at least one English letter and one number, and must not contain the word 'admin'.
+    if (!/[a-zA-Z]/.test(fsxAdminPassword) || !/\d/.test(fsxAdminPassword) || /admin/i.test(fsxAdminPassword)) {
+      throw new Error(
+        "'fsxAdminPassword' must contain at least one English letter and one number, and must not contain the word 'admin'",
+      );
+    }
+  }
+
+  private validateHaPairs(deploymentType: OntapDeploymentType, haPairs?: number): void {
+    if (haPairs == null || Token.isUnresolved(haPairs)) {
+      return;
+    }
+    if (!Number.isInteger(haPairs) || haPairs < 1 || haPairs > 12) {
+      throw new Error(`\'haPairs\' must be an integer between 1 and 12, got ${haPairs}`);
+    }
+    if (
+      haPairs > 1 &&
+      [OntapDeploymentType.SINGLE_AZ_1, OntapDeploymentType.MULTI_AZ_1, OntapDeploymentType.MULTI_AZ_2].includes(
+        deploymentType,
+      )
+    ) {
+      throw new Error(`\'haPairs\' must be 1 for deployment type ${deploymentType}`);
+    }
+  }
+
+  private validateSubnets(
+    deploymentType: OntapDeploymentType,
+    vpcSubnets: aws_ec2.ISubnet[],
+    preferredSubnet?: aws_ec2.ISubnet,
+  ): void {
+    if (
+      (deploymentType === OntapDeploymentType.MULTI_AZ_1 || deploymentType === OntapDeploymentType.MULTI_AZ_2) &&
+      !preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' must be specified for multi-AZ file systems");
+    }
+    if (
+      (deploymentType === OntapDeploymentType.SINGLE_AZ_1 || deploymentType === OntapDeploymentType.SINGLE_AZ_2) &&
+      preferredSubnet
+    ) {
+      throw new Error("'preferredSubnet' can only be specified for multi-AZ file systems");
+    }
```

Comment:
`preferredSubnet` must be specified for Multi-AZ file system.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-fsx-filesystem-ontapconfiguration.html#cfn-fsx-filesystem-ontapconfiguration-preferredsubnetid